src/index.ts

```diff
@@ -161,14 +161,20 @@ export default class Kumulos {
 
     async pushRegister(): Promise<void> {
         const pushManager = await getPushOpsManager(this.context);
-
         const permission  = await pushManager.requestNotificationPermission(this.context);
 
         if (permission !== 'granted') {
             return Promise.reject(
                 'Notification permission not granted'
             );
         }
+
+        //TODO: The below code is a hack in place to avoid an issue with the onPermissionChange event not firing from Safari: https://bugs.webkit.org/show_bug.cgi?id=256201#c1    
```

Comment:
this bug only exists in safari 16.4? does it mean in older will be called twice?

Reply:
Updated to check for presence of PushManager as well as Safari, to ensure this applies only to Safari versions that support push manager, and not legacy versions.

Reply:
zzz, have to carefully check what's supported what's not supported. This line

`const permissionStatus = await navigator.permissions.query({name: 'notifications'});`

permissions API is only available in safari 16. Then different browsers support different 'name' in query, and Safari doesnt support `notifications` at all. https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API#browser_compatibility . Maybe this explains why event is not working, so, it's not a bug? Does it mean there is an error in `observePermissionStatus` in every version of safari?

Reply:
but in any case it's not working with <16? suppose there was a reason to add listener vs always registering when promise resolved @k-antipochkin ?

Reply:
I did some additional digging on this on safari 15.X on saucelab (thanks Amir!). Looks like we are indeed failing to register for the onChange event of the notifications API permissions because the permissions api itself doesn't exist on the navigator. 

<img width="1148" alt="Screenshot 2023-05-23 at 14 41 25" src="https://github.com/Kumulos/KumulosSdkWeb/assets/41353756/b62b79a2-3b4e-4273-98a7-f2f6af75614a">

Further to this, I removed the code in here that calls the pushRegister() method and tested whether we receive anything in the optimove backend. We do NOT in fact see anything besides the firstOpened event, even though we prompted the user for consent. @ftcvlad was absolutely correct that in legacy safari push notification scenarios, we do NOT register correctly, and never send any notifications currently. 

The code we had here originally where the condition was only browser === 'safari' actually allowed both legacy and W3C push notifications to fire correctly. This doesn't, however, change the fact we have an error message in the console when in legacy mode. Ideally, where we try to get the permissions object using the permissions api, we should first check for its presence like `if (navigator.permissions && navigator.permissions.query)`. 

Tagging @cgwyllie and @k-antipochkin for visibility.

Reply:
From https://developer.apple.com/documentation/safari-release-notes/safari-16_4-release-notes

> Added fingerprinting countermeasures for querying the permission state of the Notifications API.